controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -156,6 +158,50 @@ public String getScopeName() {
         return deleteStream();
     }
 
+    /**
+     * Create a subscribersRecord with provided subscriber for this Stream.
+     * @return future of operation.
+     */
+    @Override
+    public CompletableFuture<Void> createSubscribersRecord(String subscriber) {
```

Comment:
we can maybe always create this subscriber record for a stream as part of stream creation.. 
that way it only needs to be updated if a subscriber registers or unregisters.

Reply:
The reason I did not want to do it is because a lot of Streams may never need CBR, in which case that additional metadata is an unnecessary overhead that they can do without.
Also for existing Pravega deployments, that move to the new version with CBR this metadata would need to be explicitly added when the Stream moves to CBR.